src/domino.py

```diff
@@ -0,0 +1,208 @@
+import json
+import shutil
+from pathlib import Path
+
+import pandas as pd
+
+from src.prm import PRM
+from src.util import prepare_volume, run_container
+
+__all__ = ['DOMINO']
+
+ID_PREFIX = 'ENSG0'
+
+class DOMINO(PRM):
+    required_inputs = ['network', 'active_genes']
+
+    @staticmethod
+    def generate_inputs(data, filename_map):
+        """
+        Access fields from the dataset and write the required input files
+        @param data: dataset
+        @param filename_map: a dict mapping file types in the required_inputs to the filename for that type
+        @return:
+        """
+        for input_type in DOMINO.required_inputs:
+            if input_type not in filename_map:
+                raise ValueError(f"{input_type} filename is missing")
+
+        # Get active genes for node input file
+        if data.contains_node_columns('active'):
+            #NODEID is always included in the node table
+            node_df = data.request_node_columns(['active'])
+        else:
+            raise ValueError("DOMINO requires active genes")
+        node_df = node_df[node_df['active'] == True]
+
+        # transform each node id with a prefix
+        node_df['NODEID'] = node_df['NODEID'].apply(pre_domino_id_transform)
+        # e.g., ENSG0[node_id]
+
+        #Create active_genes file
+        node_df.to_csv(filename_map['active_genes'],sep="\t",index=False,columns=['NODEID'], header=False)
+
+        #Create network file
+        edges_df = data.get_interactome()
+        edges_df['ppi'] = 'ppi'
+
+        # transform each node id with a prefix
+        edges_df['Interactor1'] = edges_df['Interactor1'].apply(pre_domino_id_transform)
+        edges_df['Interactor2'] = edges_df['Interactor2'].apply(pre_domino_id_transform)
+
+        edges_df.to_csv(filename_map['network'],sep='\t',index=False,columns=['Interactor1','ppi','Interactor2'],header=['ID_interactor_A','ppi','ID_interactor_B'])
+
+
+
+    @staticmethod
+    def run(network=None, active_genes=None, output_file=None, use_cache=True, slices_threshold=None, module_threshold=None, singularity=False):
+        """
+        Run DOMINO with Docker
+        Let visualization be always true, parallelization be always 1 thread
```

Comment:
This is another good place to document how the multiple output modules will be merged into a single pathway.